tests/wicked/sysctl/sut/t01_sysctl_load_files.pm

```diff
@@ -0,0 +1,149 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Check the sysctl.d file load order of wicked. It's not like
+#          `sysctl --system` as procps-ng >=3.3.17 changed the
+#          sysctl.d load order like systemd.
+#
+# Maintainer: cfamullaconrad@suse.com
+
+use Mojo::Base 'wickedbase';
+use testapi;
+use autotest ();
+use File::Basename;
+use Mojo::Util qw(trim);
+
+our $wicked_show_config = 'wicked --log-level debug --debug all  show-config all';
+our @sysctl_d = qw(
+  /run/sysctl.d
+  /etc/sysctl.d
+  /usr/local/lib/sysctl.d
+  /usr/lib/sysctl.d
+  /lib/sysctl.d
+);
+
+sub wicked_get_file_order {
+    my $out = script_output($wicked_show_config . ' |& grep "Reading sysctl file"');
+    my @files = ($out =~ m/file\s+'([^']+)'/g);
+    return \@files;
+}
+
+sub wicked_get_file_errors {
+    my $out = script_output($wicked_show_config . ' |& grep "Cannot open"');
+    my @files = ($out =~ m/Cannot open\s+'([^']+)'/g);
+    return \@files;
+}
+
+sub sysctl_emu_file_order {
+    state $boot_sysctl = '/boot/sysctl.conf-' . trim(script_output('uname -r'));
+    my @retval = ($boot_sysctl);
+    my @sysctl_d_files = split(/\r?\n/, script_output("find @sysctl_d -name '*.conf' 2> /dev/null", proceed_on_failure => 1));
+
+    my %sysctl_d;
+    for my $file (@sysctl_d_files) {
+        my $basename = basename($file);
+        $sysctl_d{$basename} = $file unless $sysctl_d{$basename};
+    }
+
+    push @retval, $sysctl_d{$_} foreach (sort keys %sysctl_d);
+    push @retval, '/etc/sysctl.conf' if script_run('test -e /etc/sysctl.conf', die_on_timeout => 1) == 0;
+
+    return \@retval;
+}
+
+sub check_load_order {
+    my $exp_order = shift // sysctl_emu_file_order();
+    my $wicked = wicked_get_file_order;
+
+    unless (join(',', @$exp_order) eq join(',', @$wicked)) {
+        die("wicked load the sysctl files in different order\n" .
+              "expect: @$exp_order\n" .
+              "wicked: @$wicked\n");
+    }
+}
+
+sub check {
+    my ($self, $ctx) = @_;
+
+    check_load_order();
+
+    # Check order of files in sysctl.d directories
+    for my $dir (@sysctl_d) {
+        assert_script_run("mkdir -p $dir && touch $dir/20-test.conf");
```

Comment:
I think you can move `mkdir -p` just before this two loops and not waste time on typing this command over and over again